subprojects/docs/src/samples/java/jvm-multi-project-with-code-coverage/kotlin/buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts

```
@@ -51,9 +51,9 @@ configurations.create("coverageDataElements") {
         attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.DOCUMENTATION))
         attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named("jacoco-coverage-data"))
     }
-    // This will cause the test task to run if the coverage data is requested by the aggregation task
-    outgoing.artifact(tasks.test.map { task ->
-        task.extensions.getByType<JacocoTaskExtension>().destinationFile!!
-    })
+    // This will cause test tasks to run if the coverage data is requested by the aggregation task
+    tasks.withType(Test::class) {
```

Could you use _withType<Test>()_ here, which is the more _idiomatic_ way of addressing types in Kotlin (if the API allows).

And it has to be `withType<Test>().configureEach { .. }` to avoid eager configuration of all test tasks. Similar as the Groovy equivalent. Then you also need to keep the `map { ... }`

```suggestion
    tasks.withType<Test>().configureEach {
```

---

I've always tried this templated form of withType to no success but may have been missing the parens - thanks for sharing!

For configureEach I may have had a very incorrect model for kotlin dsl. In docs

https://docs.gradle.org/current/userguide/kotlin_dsl.html

> Note that accessors for elements of containers such as configurations, tasks and sourceSets leverage Gradle’s configuration avoidance APIs.

I assumed this meant all kotlin usage is lazy even without configureEach but must have misunderstood 😨